### ISSUE TITLE
Can't use aws_lambda_layer_version in lambda declaration

### ISSUE THREAD

**Issue by mateusfccp** [terraform-providers/terraform-provider-aws#7728]
I don't know if this issue is related to the core terraform or this provider, but I posted here because all other resources works correctly.

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.11.11
+ provider.archive v1.1.0
+ provider.aws v1.60.0

### Affected Resource(s)

* aws_lambda_layer_version

### Terraform Configuration Files

```hcl
# Core layer
resource "aws_lambda_layer_version" "core_layer" {
  filename = "core.zip"
  layer_name = "core"
  source_code_hash = "${data.archive_file.core_zip.output_base64sha256}"

  compatible_runtimes = ["python3.7"]
}

resource "aws_lambda_function" "lambda" {
  [...]
  layers           = ["${aws_lambda_layer_version.core_layer.lambda_arn}"]
}

```

### Expected Behavior

My lambda function should accept the resources arns to be used on `layers` variable

### Actual Behavior

When trying to `terraform apply`:

```
Error: resource 'aws_lambda_function.lambda' config: unknown resource 'aws_lambda_layer_version.core_layer' referenced in variable aws_lambda_layer_version.core_layer.lambda_arn
```

### Steps to Reproduce

1. Refer to `aws_lambda_layer_version` resource on a lambda resource  `layers` variable

2. `terraform apply`

**Comment by mateusfccp** [comment]
This is also happening on AWS provider version 2.2.0, as I tested today.

**Comment by mateusfccp** [comment]
It was fixed today. You may close.

**Comment by ericfrederich** [comment]
@mateusfccp has this fix made it into a release?

**Comment by mateusfccp** [comment]
I'm not sure, but I think so, as the error stopped suddenly, at least for me.

**Comment by aeschright** [comment]
Fixed in v2.3.0
